plugins/device/deviceutils/store-utils.go

```diff
@@ -28,12 +28,11 @@ func GetDevices(peerIds ...string) ([]deviceapi.Info, error) {
 			return nil, err
 		}
 	}
-
 	var devices []deviceapi.Info
 	for _, peerInfo := range peers {
-		var deviceInfo []deviceapi.Info
-		if len(peerInfo.Metadata["_devices"]) > 0 {
-			if err := json.Unmarshal([]byte(peerInfo.Metadata["_devices"]), &deviceInfo); err != nil {
+		if _, exists := peerInfo.Metadata["_devices"]; exists {
```

Comment:
This check will be done within the function and isn't necessary here. The consumer only has to check if the devices list is empty and not other about how it's stored.